### DISCUSSION TITLE
Any way to have separated physics scenes or to manually simulate physics?

### DISCUSSION THREAD

**Discussion by Awarets** [stride3d/stride#1866] [Q&A]
Is there any way to have a set of physics objects operate independently from other objects, and is there any way to disable automatic simulation of this set, and instead manually simulate this set through script?

This is necessary for my network code as I need to be able to roll back the state of the game, correct it, and replay it in order to resolve any inaccuracies from clientside input prediction.

This was possible for me in Unity using LocalPhysicsMode.Physics3D and Physics.Simulate(), and I'm wondering if Stride allows for any equivalent functionality.

Edit: After investigating further, I've found that Jolt Physics allows for the overall state of the physics scene to be saved, restored (from that saved state), and manually updated. 
If a similar solution to this is possible in Stride, this would also be suitable. It would actually be preferable, since it eliminates the need for all physics objects to be duplicated into a separate scene.

**Comment by manio143** [comment]
It might not be as straightforward.
The pieces that would make this possible are kind of there, but there's currently a bit of coupling.
So the way physics work is that a SceneInstance manges a set of entities with physics components and has an associated Simulation in the physics system. You could potentially create a second SceneInstance and call Update on it manually, but the Simulation itself is updated from the main game loop.
Another option would be to create a whole second Game instance with a separate update loop (that can be stepped manually). You might need to create your own Game class with just scene and physics and remove scripts, graphics, etc.
And then you would sync your entities across those two game instances.

**Comment by lucas-0liveira** [comment]
In Stride, you can achieve independent physics simulation for a specific set of objects by implementing a custom simulation step through scripting. This allows you to disable automatic simulation for this set and manually handle forces and interactions. While not a direct match to Unity's LocalPhysicsMode.Physics3D and Physics.Simulate(), this approach enables you to roll back and replay game states for network synchronization.
